### PR TITLE
Adding cycle 2 to curriculum visualiser behind flag

### PR DESCRIPTION
**Pull request by orangemug**
## Description
Added cycle 2 data into the visualiser behind a flag

## Issue(s)
`CUR-800`

## How to test

1. Go to https://deploy-preview-2692--oak-web-application.netlify.thenational.academy/teachers/curriculum
2. Check for regressions

Enable cycle 2 locally and test we're pulling though the data. Probably some issues at this point, we can raise further tickets as long as the data is essentially there. 

## Screenshots
Screenshot 2024-08-13 at 17 20 04 Screenshot 2024-08-13 at 17 19 42

### CODE REVIEW

File: src/components/CurriculumComponents/helpers/curriculumMetadata.ts [hunk 1]
```diff
@@ -19,31 +19,40 @@ function getExamboardFromSlug(props: buildCurriculumMetadataProps) {
...
   props: buildCurriculumMetadataProps,
 ) => {
+  const { keyStages } = props;
+  const keyStagesItems: string[] = [];
+  if (keyStages.includes("ks1")) keyStagesItems.push("1");
+  if (keyStages.includes("ks2")) keyStagesItems.push("2");
+  if (keyStages.includes("ks3")) keyStagesItems.push("3");
+  if (keyStages.includes("ks4")) keyStagesItems.push("4");
+  const keyStagesData =
+    keyStagesItems.length > 0 ? `K${keyStagesItems.join("-")}` : ``;
```
Comment (rightsaidjames) [inline src/components/CurriculumComponents/helpers/curriculumMetadata.ts:29]: It probably doesn't matter if this is just a spike branch, but just to note that this should be:
```suggestion
    keyStagesItems.length > 0 ? `KS${keyStagesItems.join("-")}` : ``;
```

File: src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx [hunk 2]
```diff
@@ -544,12 +552,28 @@ export const getStaticProps: GetStaticProps<
...
 
       const subjectPhaseOptions = await fetchSubjectPhasePickerData();
+
       const results: GetStaticPropsResult<CurriculumInfoPageProps> = {
         props: {
           curriculumSelectionSlugs: slugs,
           subjectPhaseOptions,
           curriculumOverviewTabData,
-          curriculumOverviewSanityData,
+          // TODO: Hack to null the object for testing
```
Comment (orangemug) [inline src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx:561]: We need to merge the new CMS branch prior to removing this (CC @sonalisin)

File: src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx [hunk 3]
```diff
@@ -515,12 +517,18 @@ export const getStaticProps: GetStaticProps<
           ...slugs,
         });
 
-      if (!curriculumOverviewSanityData) {
-        return {
-          notFound: true,
-        };
-      }
-      const curriculumUnitsTabData = await curriculumApi.curriculumUnits(slugs);
+      // if (!curriculumOverviewSanityData) {
```
Comment (orangemug) [inline src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx:520]: Same issue as https://github.com/oaknational/Oak-Web-Application/pull/2692/files#r1762845424

File: src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx [hunk 4]
```diff
@@ -515,12 +517,18 @@ export const getStaticProps: GetStaticProps<
...
+      //   return {
+      //     notFound: true,
+      //   };
+      // }
+      const querySlugs = {
+        ...slugs,
+        state: "new",
```
Comment (orangemug) [inline src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx:527]: Make configurable based on global constant to enable "cycle 2"

File: src/components/CurriculumComponents/CurriculumHeader/CurriculumHeader.tsx [hunk 5]
```diff
@@ -59,12 +61,18 @@ const CurriculumHeader: FC<CurriculumHeaderPageProps> = ({
...
 
   let pageTitle: string;
+  const keyStageStrings: string[] = [];
+  if (keyStages.includes("ks1")) keyStageStrings.push("KS1");
+  if (keyStages.includes("ks2")) keyStageStrings.push("KS2");
+  if (keyStages.includes("ks3")) keyStageStrings.push("KS3");
+  if (keyStages.includes("ks4")) keyStageStrings.push("KS4");
+  const keyStageString = keyStageStrings.join(" & ");
   switch (phase.slug) {
     case "primary":
```
Comment (sonalisin) [inline src/components/CurriculumComponents/CurriculumHeader/CurriculumHeader.tsx:71]: Do we still need the case if the`pageTitle` is the same for both cases?

**Review by sonalisin — APPROVED** [review]
All good

**Review by rightsaidjames — APPROVED** [review]
Tested for regressions and to ensure that no cycle 2 data was leaking through. All good!

File: src/components/CurriculumComponents/CurriculumHeader/CurriculumHeader.tsx (hunk 5 above)
Comment (orangemug) [reply src/components/CurriculumComponents/CurriculumHeader/CurriculumHeader.tsx:71]: Added ticket to fix in `CUR-923` (just so we can get this merged)

**Pull request merged**